### ISSUE TITLE
char support in C++ grids?

### ISSUE THREAD

**Issue by diregoblin** [multiscale/muscle3#61]
Sometimes it's useful to send a grid of values in a relatively small range, e.g. cell types in the ISR model. Converting everything into ``int``'s feels like an unnecessary complication in these cases. Would it be reasonable to include a single-byte type specification for ``Grid``?

**Comment by LourensVeen** [comment]
Yep, I think that this is a valid use case. I guess alternatively there could be a constructor that takes a `char *` and creates a grid of `int32_t`, but that makes no sense. I'll add one. I've used unsigned ints in programs a few times, and I've always ended up regretting it, so we'll stick with `int8_t`. If the numbers are larger than 127 then you'll have to move a step up. I think `int16_t` is pretty rare, so let's leave that out for now and hope for the best.

- [ ] Python: Add ExtTypeId for grid-of-int8
- [ ] Python: Add encoder, decoder and unit tests
- [ ] C++: Add ExtTypeId for grid-of-int8
- [ ] C++: Add encoder, decoder and unit tests
- [ ] C++: Add Python-C++ integration test
- [ ] Fortran: Add wrappers to API
- [ ] Fortran: Add API test
